src/RepayRewardEscrow.sol

```diff
@@ -0,0 +1,214 @@
+        require(lsInvOut >= expectedShares, "Insufficient shares received");
+
+        _mint(msg.sender, lsInvOut);
+
+        dolaAllocations[msg.sender] = 0;
+
+        emit Buy(msg.sender, dolaAmountIn, invAmount, lsInvOut);
+    }
+
+    /**
+     * @notice Allows users to redeem their lsINV shares for sINV tokens after the redemption period.
+     * @dev Users must wait until the redemption timestamp to redeem their shares.
+     * @param lsInvAmount The number of lsINV shares to redeem.
+     */
+    function redeem(uint256 lsInvAmount) external {
+        require(block.timestamp >= redemptionTimestamp && redemptionTimestamp != 0, "Redemption not started yet");
+        require(lsInvAmount > 0, "lsInvAmount must be greater than zero");
+
+        _burn(msg.sender, lsInvAmount);
+
+        sINV.safeTransfer(msg.sender, lsInvAmount);
+
+        emit Redeem(msg.sender, lsInvAmount);
+    }
+
+    /**
+     * @notice Allows anyone to send DOLA to the sale handler for debt repayment.
+     * Will send up to the Sale Handler capacity.
+     */
+    function sendToSaleHandler() external {
+        uint256 bal = DOLA.balanceOf(address(this));
+        require(bal > 0, "No DOLA to send");
+        uint256 capacity = SALE_HANDLER.getCapacity();
+        uint256 amount = bal > capacity ? capacity : bal;
+        DOLA.transfer(address(SALE_HANDLER), amount);
+        SALE_HANDLER.onReceive();
+        emit DolaRepayment(amount);
+    }
+
+    // Admin functions
+
+    /**
+     * @notice Sets a allocation for a user on how much DOLA they can spend to buy lsINV.
+     * @dev Only the operator can set DOLA allocations. This is used to control the amount of INV each user can purchase.
+     * @param user The address of the user to set the allocation for.
+     * @param allocation The amount of DOLA the user is allowed to spend.
+     */
+    function setDolaAllocation(address user, uint256 allocation) external onlyOperator {
+        dolaAllocations[user] = allocation;
+        emit DolaAllocationSet(user, allocation);
+    }
+
+    /**
+     * @notice Extends the buy deadline by a specified number of days.
+     * @dev This function can only be called by the operator. It allows the operator to extend the buy period if needed.
+     * @param extraDays The number of days to extend the buy deadline by.
+     */
+    function extendDeadline(uint256 extraDays) external onlyOperator {
+        uint256 newDeadline = block.timestamp + (extraDays * 1 days);
+        require(newDeadline > buyDeadline, "New deadline must be later than current");
+        buyDeadline = newDeadline;
+        emit BuyDeadlineUpdated(buyDeadline);
+    }
+
+    /**
+     * @notice Starts the buy period by setting the buy deadline to 4 days and the redemption timestamp to 6 months from now.
+     * @dev This function can only be called by governance. It allows users to start buying lsINV tokens.
+     * The buy period lasts for 4 days.
+     */
+    function start() external onlyGov {
```

Comment:
start can be called multiple times by gov. Technically it could allow governance to delay the redemption deadline for lsINV holders